src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -3,264 +3,150 @@
+            return $"{description}{(description.EndsWith(".") ? "" : ".")} Allowed values: {BuilderHelpers.EscapeXmlDocDescription(allowedValues)}";
         }
 
-        private static Constant? ParseConstant(RequestParameter parameter, TypeFactory typeFactory)
+        public static Validation GetValidation(CSharpType type, RequestLocation requestLocation, bool skipUrlEncoding)
         {
-            if (parameter.In == HttpParameterIn.Header && RequestHeader.ClientRequestIdHeaders.Contains(parameter.Language.Default.SerializedName ?? parameter.Language.Default.Name))
-            {
-                return Constant.FromExpression($"message.{Configuration.ApiTypes.HttpMessageRequestName}.ClientRequestId", new CSharpType(typeof(string)));
-            }
-            if (parameter.In == HttpParameterIn.Header && RequestHeader.ReturnClientRequestIdResponseHeaders.Contains(parameter.Language.Default.SerializedName ?? parameter.Language.Default.Name))
+            if (requestLocation is RequestLocation.Uri or RequestLocation.Path or RequestLocation.Body && type.EqualsIgnoreNullable(typeof(string)) && !skipUrlEncoding)
             {
-                return new Constant("true", new CSharpType(typeof(string)));
+                return Validation.AssertNotNullOrEmpty;
             }
-            if (parameter.Schema is ConstantSchema constantSchema && parameter.IsRequired)
+
+            if (!type.IsValueType)
             {
-                return BuilderHelpers.ParseConstant(constantSchema.Value.Value, typeFactory.CreateType(constantSchema.ValueType, constantSchema.Value.Value == null));
+                return Validation.AssertNotNull;
             }
 
-            return null;
+            return Validation.None;
         }
 
         public static readonly IEqualityComparer<Parameter> EqualityComparerByType = new ParameterByTypeEqualityComparer();
-        private struct ParameterByTypeEqualityComparer : IEqualityComparer<Parameter>
+        private class ParameterByTypeEqualityComparer : IEqualityComparer<Parameter>
         {
-            public bool Equals(Parameter? x, Parameter? y)
-            {
-                return Object.Equals(x?.Type, y?.Type);
-            }
-
-            public int GetHashCode([DisallowNull] Parameter obj) => obj.Type.GetHashCode();
+            public bool Equals(Parameter? x, Parameter? y) => Equals(x?.Type, y?.Type);
+            public int GetHashCode(Parameter obj) => obj.Type.GetHashCode();
         }
     }
 
+    internal record Validation(ValidationType Type, object? Data)
+    {
+        public static Validation None { get; } = new(ValidationType.None, null);
+        public static Validation AssertNotNull { get; } = new(ValidationType.AssertNotNull, null);
+        public static Validation AssertNotNullOrEmpty { get; } = new(ValidationType.AssertNotNullOrEmpty, null);
+    }
+
     internal enum ValidationType
     {
         None,
+        AssertNull,
```

Comment:
is this value used anywhere? Do we really have checks to check if a parameter is null?

Reply:
Yes, in HLC